packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -112,30 +125,43 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
             {
                 eventName: "CreateNew",
                 isWithSummaryUpload: true,
-                createShareLinkParam: JSON.stringify(createShareLinkParam),
+                createShareLinkParam: createShareLinkParam ? JSON.stringify(createShareLinkParam) : undefined,
                 enableShareLinkWithCreate: this.hostPolicy.enableShareLinkWithCreate,
                 enableSingleRequestForShareLinkWithCreate:
                     this.hostPolicy.enableSingleRequestForShareLinkWithCreate,
             },
             async (event) => {
-                odspResolvedUrl = await createNewFluidFile(
-                    toInstrumentedOdspTokenFetcher(
-                        odspLogger,
-                        resolvedUrlData,
-                        this.getStorageToken,
-                        true /* throwOnNullToken */,
-                    ),
-                    newFileInfo,
+                const getStorageToken = toInstrumentedOdspTokenFetcher(
```

Comment:
There is a change in behavior here - previously token would be fetched within createNewFluidFile().
I believe previous behavior was correct and new one is not.
Consider the case that host returns cached token that is already expired or will expire by the time we make call to SPO. In previous design we would turn around and ask host for new refreshed token, with current code we would fail right away.